src/raster/rasterband.rs

```diff
@@ -863,6 +945,55 @@ pub struct StatisticsAll {
     pub std_dev: f64,
 }
 
+#[derive(Debug)]
+pub struct Histogram {
+    min: f64,
+    max: f64,
+    n_buckets: usize,
+    counts: HistCounts,
+}
+
+impl Histogram {
+    /// Histogram lower bound
+    pub fn min(&self) -> f64 {
+        self.min
+    }
+
+    /// Histogram upper bound
+    pub fn max(&self) -> f64 {
+        self.max
+    }
+    /// Histogram values for each bucket
+    pub fn counts(&self) -> &[i32] {
+        match &self.counts {
+            HistCounts::Borrowed(p) => unsafe { std::slice::from_raw_parts(*p, self.n_buckets) },
+            HistCounts::Owned(v) => v.as_slice(),
+        }
+    }
+
+    /// Histogram bucket size
+    pub fn bucket_size(&self) -> f64 {
+        (self.max - self.min) / self.counts().len() as f64
+    }
+}
+
+#[derive(Debug)]
+enum HistCounts {
+    Borrowed(*mut i32),
+    Owned(Vec<i32>),
+}
+
+impl Drop for HistCounts {
+    fn drop(&mut self) {
+        match self {
+            HistCounts::Borrowed(p) => unsafe {
+                gdal_sys::VSIFree(*p as *mut libc::c_void);
+            },
```

Comment:
@spadarian This is where I'm still confused.... if it were truly borrowed, it means it's owned by the C runtime, and we shouldn't be freeing it? That's what I meant earlier about not being sure if I had the owned vs borrowed cases correct.

Reply:
I don't know... not an expert in these things. The C documentation says: "@param ppanHistogram pointer to array into which the histogram totals are placed. To be freed with VSIFree" and we expose the `VSIFree` function, so I assumed that it is a way of saying to the C runtime that we don't need that anymore, which is what happen when we drop it.

If it is not needed, I can remove it.

Reply:
Great, that helps. It's needed, but the names `Borrowed` and `Owned` need to be something else. I'll try to make time today to come up with an alternative proposal. Need to spend some time with the docs for `GDALGetRasterHistogram` and `GDALGetDefaultHistogram` first. I think you're close here.